community/rfcs/techniques/design/20200306-weight-clustering-design.md

```diff
@@ -0,0 +1,144 @@
+
+*NOTE: This proposal does not affect any of the existing pruning functionality and
+is completely independent from the rest of the existing code base.*
+
+The clustering module is structured as follows:
+
+* [cluster.py](../../tensorflow_model_optimization/python/core/clustering/keras/cluster.py) and [cluster_test.py](../../tensorflow_model_optimization/python/core/clustering/keras/cluster_test.py) &mdash; high level clustering interfaces and tests;
+* [cluster_wrapper.py](../../tensorflow_model_optimization/python/core/clustering/keras/cluster_wrapper.py) and [cluster_wrapper_test.py](../../tensorflow_model_optimization/python/core/clustering/keras/cluster_wrapper_test.py) &mdash; implementation of a particular clustering algorithm and associated tests;
+* [clusterable_layer.py](../../tensorflow_model_optimization/python/core/clustering/keras/clusterable_layer.py) &mdash; similar to `PrunableLayer` &mdash; provides an abstract base class for custom layers to be recognized as clusterable;
+* [clustering_centroids.py](../../tensorflow_model_optimization/python/core/clustering/keras/clustering_centroids.py) and [clustering_centroids_test.py](../../tensorflow_model_optimization/python/core/clustering/keras/clustering_centroids_test.py) &mdash; implementations of different cluster centroid initialization techniques and tests for each of them;
+* [clustering_registry.py](../../tensorflow_model_optimization/python/core/clustering/keras/clustering_registry.py) and [clustering_registry_test.py](../../tensorflow_model_optimization/python/core/clustering/keras/clustering_registry_test.py) &mdash; similar to
+`PruningRegistry` &mdash; contains the list of clusterable parameters in each
+layer.
+
+## Clustering Workflow
+
+In general, the clustering workflow consists of several distinct steps, as follows:
+
+1. **Clustering** &mdash; run clustering on a trained model by invoking the `cluster_weights()` function with the desired clustering parameters:
+
+    ```python
+    clustered_model = cluster.cluster_weights(model, **clustering_params)
+    ```
+
+    NOTE: Clustering happens as part of the model building process, therefore in order to cluster sequential models, we need to call `model.build()` explicitly after applying `cluster_weights()` &mdash; see [the example](#example-0-cluster-an-entire-model) below.
+
+2. **Fine-tuning** &mdash; train the clustered model to fine-tune the initial values of cluster centroids:
+
+   ```python
+   clustered_model.fit(x_training_data, y_training_data, epochs=1)
+   ```
+
+3. **Stripping** &mdash; call `strip_clustering()` to strip the clustering-specific information from the model and re-obtain the original model structure:
+
+   ```python
+   stripped_model = cluster.strip_clustering(clustered_model)
+   ```
+
+The code snippets below illustrate common usage patterns of the clustering API. For the sake of brevity, the clustering examples presented here only deal with the 1st step of the above workflow.
```

Comment:
You can generally keep this content, but have a sentence at the top that references the Tutorials & Examples from the [pruning RFC](https://github.com/tensorflow/model-optimization/pull/268/files?short_path=edfd107#diff-edfd10773bb6bb7662ce4fb0b94d12a1) and say something like "Everything is the same, except for (list the differences)"